pkg/controller/template/render.go

```diff
@@ -99,6 +99,7 @@ func GenerateMachineConfigsForRole(config *RenderConfig, role string, path strin
 	var tempIgnConfig ignv2_2types.Config
 	tempUser := ignv2_2types.PasswdUser{Name: "core", SSHAuthorizedKeys: []ignv2_2types.SSHAuthorizedKey{ignv2_2types.SSHAuthorizedKey(config.SSHKey)}}
 	tempIgnConfig.Passwd.Users = append(tempIgnConfig.Passwd.Users, tempUser)
+	tempIgnConfig.Ignition.Version = "2.2.0"
```

Comment:
I know this is WIP but I've looked at the issue some days ago as well and we definitely need a single place where we build the machine-config skeleton with the version we want and when doing ssh/kubelet/containerruntime stuff we generate the skeleton with the version once and set every other piece we need on a case by case (if it doesn't make sense I can further explain)

Reply:
yeah the ssh key would be straightforward for that, but still investigating how the kubelet/containerruntime is made since we didnt really write those. going to try to just get this fundamentally working first in the draft then optimise (if it makes sense).

Reply:
i have some ideas for a refactor that i'd like to discuss, but since this is needed for another PR (and works as is and contains an ssh refactor that aligns it with how the files/unit igns are done) I'd like to get this approved, get the mcc ignition check in to close 502 then come back to it.  wdyt @runcom  ?

Reply:
also ci is finally green and i dont know when that will happen again.